src/components/Search/index.tsx

```diff
@@ -143,15 +144,14 @@ function Search({query, policyIDs, sortBy, sortOrder, isMobileSelectionModeActiv
             SearchActions.createTransactionThread(hash, item.transactionID, reportID, item.moneyRequestReportActionID);
         }
 
-        Navigation.navigate(ROUTES.SEARCH_REPORT.getRoute(query, reportID));
+        Navigation.navigate(ROUTES.SEARCH_REPORT.getRoute(reportID));
     };
 
     const fetchMoreResults = () => {
         if (!searchResults?.search?.hasMoreResults || shouldShowLoadingState || shouldShowLoadingMoreItems) {
             return;
         }
-        const currentOffset = searchResults?.search?.offset ?? 0;
-        SearchActions.search({hash, query, offset: currentOffset + CONST.SEARCH.RESULTS_PAGE_SIZE, sortBy, sortOrder});
+        setOffset(offset + CONST.SEARCH.RESULTS_PAGE_SIZE);
```

Comment:
We increased the offset here but we forgot to reset it when we changed the tab, once offset was increased here and we changed the tab, it kept the old offset which led to the wrong data arriving.
More details:
https://github.com/Expensify/App/issues/55235#issuecomment-2591253857